torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1086,8 +1086,15 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
 
   {
     at::cuda::CUDAMultiStreamGuard streamGuard(ncclStreams_[key]);
+    std::vector<c10::DeviceIndex> deviceIndices;
+    deviceIndices.reserve(devices.size());
+    for (const at::Device& device : devices) {
+      TORCH_INTERNAL_ASSERT(device.is_cuda());
+      deviceIndices.push_back(device.index());
+    }
     work->future_ = c10::make_intrusive<at::cuda::CUDAFuture>(
-        c10::ListType::create(c10::TensorType::get()));
+        c10::ListType::create(c10::TensorType::get()),
+        std::move(deviceIndices));
```

Comment:
I think there is also a call to cuda future in ` ProcessGroupNCCL::pointToPoint`, do you want to add those changes there too?

Reply:
Good catch, thanks!